index.ts

```diff
@@ -316,68 +362,87 @@ class ServerlessCustomDomain {
-            return new DomainInfo(domainInfo);
-        } catch (err) {
-            this.logIfDebug(err);
-            if (err.code === "NotFoundException") {
-                throw new Error(`Error: ${this.givenDomainName} not found.`);
+    public async getDomainInfo(): Promise<void> {
+        await Promise.all(this.domains.map(async (domain) => {
+            try {
+                const domainInfo = await this.apigatewayV2.getDomainName({
+                    DomainName: domain.givenDomainName,
+                }).promise();
+
+                domain.domainInfo = new DomainInfo(domainInfo);
+            } catch (err) {
+                this.logIfDebug(err, domain.givenDomainName);
+                if (err.code !== "NotFoundException") {
+                    throw new Error(`Error: Unable to fetch information about ${domain.givenDomainName}`);
+                }
             }
-            throw new Error(`Error: Unable to fetch information about ${this.givenDomainName}`);
-        }
+        }));
     }
 
     /**
      * Creates Custom Domain Name through API Gateway
      * @param certificateArn: Certificate ARN to use for custom domain
      */
-    public async createCustomDomain(certificateArn: string): Promise<DomainInfo> {
-        // Set up parameters
-        const params = {
-            certificateArn,
-            domainName: this.givenDomainName,
-            endpointConfiguration: {
-                types: [this.endpointType],
-            },
-            regionalCertificateArn: certificateArn,
-            securityPolicy: this.securityPolicy,
-        };
-        if (this.endpointType === endpointTypes.edge) {
-            params.regionalCertificateArn = undefined;
-        } else if (this.endpointType === endpointTypes.regional) {
-            params.certificateArn = undefined;
-        }
+    public async createCustomDomain(domain: DomainConfig): Promise<void> {
 
-        // Make API call
         let createdDomain = {};
-        try {
-            createdDomain = await this.apigateway.createDomainName(params).promise();
-        } catch (err) {
-            this.logIfDebug(err);
-            throw new Error(`Error: Failed to create custom domain ${this.givenDomainName}\n`);
+
+        // For EDGE domain name, create with APIGateway (v1)
+        if (domain.endpointType === Globals.endpointTypes.edge) {
+            // Set up parameters
+            const params = {
+                certificateArn: domain.certificateArn,
+                domainName: domain.givenDomainName,
+                endpointConfiguration: {
+                    types: [domain.endpointType],
+                },
+                securityPolicy: domain.securityPolicy,
+            };
+
+            // Make API call to create domain
+            try {
+                // If creating REST api use v1 of api gateway, else use v2 for HTTP and Websocket
```

Comment:
Is this comment still accurate? This seems to decide which API to use based on endpoint type and not API type

Reply:
Also, why can't EDGE domains/rest APIs be created using the v2 API?

Reply:
seems like it can https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/ApiGatewayV2.html#createDomainName-property

Reply:
@aoskotsky-amplify @claygorman That inline comment is incorrect and should say if creating an EDGE API (only REST can be EDGE, so it's a little confusing, but if REST is regional it can us v2 API), but.....

...the AWS docs are wrong, and not yet updated. I opened a case with AWS Support about this issue exactly. 

v1 API cannot create regional domain names.

https://github.com/amplify-education/serverless-domain-manager/pull/319#issuecomment-606131803

<img width="1023" alt="Screen Shot 2020-05-01 at 7 39 08 PM" src="https://user-images.githubusercontent.com/1007085/80849077-a0cf5200-8be3-11ea-91d5-4306b18b91dd.png">
<img width="995" alt="Screen Shot 2020-05-01 at 7 39 17 PM" src="https://user-images.githubusercontent.com/1007085/80849079-a3ca4280-8be3-11ea-961a-835304fbb0f5.png">
<img width="989" alt="Screen Shot 2020-05-01 at 7 39 23 PM" src="https://user-images.githubusercontent.com/1007085/80849083-a6c53300-8be3-11ea-8f93-ac965b8da4ff.png">
<img width="988" alt="Screen Shot 2020-05-01 at 7 39 40 PM" src="https://user-images.githubusercontent.com/1007085/80849089-aa58ba00-8be3-11ea-8068-a250c52a79ac.png">

Reply:
I think we are good to resolve this one? @aoskotsky-amplify